ibm/service/kms/resource_ibm_kms_key_rings_test.go

```diff
@@ -59,64 +63,197 @@ func TestAccIBMKMSResource_Key_Ring_Not_Exist(t *testing.T) {
 		Providers: acc.TestAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config:      testAccCheckIBMKmsResourceKeyRingExistConfig(instanceName, keyName, keyRing),
+				Config:      buildResourceSet(WithResourceKMSInstance(instanceName), WithResourceKMSKey(keyName, keyRing)),
 				ExpectError: regexp.MustCompile("KEY_RING_NOT_FOUND_ERR:"),
 			},
 		},
 	})
 }
 
-func testAccCheckIBMKmsResourceKeyRingConfig(instanceName, keyRing string) string {
-	return fmt.Sprintf(`
-	resource "ibm_resource_instance" "kms_instance" {
-		name              = "%s"
-		service           = "kms"
-		plan              = "tiered-pricing"
-		location          = "us-south"
+func TestAccIBMKMSResource_Key_Ring_ForceDeleteFalse(t *testing.T) {
+	instanceName := fmt.Sprintf("tf_kms_%d", acctest.RandIntRange(10, 100))
+	keyName := fmt.Sprintf("key_%d", acctest.RandIntRange(10, 100))
+	keyRing := fmt.Sprintf("keyRing%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { acc.TestAccPreCheck(t) },
+		Providers: acc.TestAccProviders,
+		Steps: []resource.TestStep{
+			// Create a Key Ring and check force_delete is false
+			{
+				Config: buildResourceSet(WithResourceKMSInstance(instanceName), WithResourceKMSKeyRing(keyRing, false), WithResourceKMSKey(keyName, "ibm_kms_key_rings.test.key_ring_id")),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("ibm_kms_key.test", "key_name", keyName),
+					resource.TestCheckResourceAttr("ibm_kms_key.test", "key_ring_id", keyRing),
+					resource.TestCheckResourceAttr("ibm_kms_key_rings.test", "force_delete", "false"),
+				),
+			},
+			// Developer note: We cannot move key rings to default key ring as we have not implemented that PATCH endpoint in terraform. Therefore we must depend on the force_delete flag to clean up test cases
+			// Attempt to delete the key ring and key
```

Comment:
What does "cannot move key rings to default key ring" mean?

Reply:
We have an API that can patch a key from one key ring to another. For the purposes of testing, I would've liked to do this but we do not support this in our terraform provider. Made this note so other devs dont think it is possible, since it will appear like it works when it really doesnt.

Reply:
But the idea of moving a key ring to another key ring doesn't make sense to me. Don't even know why someone would think it would be possible. But maybe that is an expectation someone might have?